src/pip/_internal/commands/list.py

```diff
@@ -357,5 +367,6 @@ def format_for_json(packages: "_ProcessedDists", options: Values) -> str:
         editable_project_location = dist.editable_project_location
         if editable_project_location:
             info["editable_project_location"] = editable_project_location
+        info["metadata"] = dist.json_metadata
```

Comment:
I kind of feel this is not worth a separate property; simply inlining `msg_to_json(dist.metadata)` could be better.

(BTW `msg_to_json` is not a very good function name.)